schedule/search.go

```diff
@@ -26,29 +35,51 @@ type SearchOptions struct {
 
 // SearchCursor is used to indicate a position in a paginated list.
 type SearchCursor struct {
-	Name string `json:"n,omitempty"`
+	Name       string `json:"n,omitempty"`
+	IsFavorite bool   `json:"f"`
 }
 
 var searchTemplate = template.Must(template.New("search").Parse(`
 	SELECT
-		id, name, description, time_zone
+		sched.id,
+		sched.name,
+		sched.description,
+		sched.time_zone,
+		fav IS DISTINCT FROM NULL
 	FROM schedules sched
+	{{if not .FavoritesOnly }}
+		LEFT {{end}}JOIN user_favorites fav ON sched.id = fav.tgt_schedule_id 
+		AND {{if .FavoritesUserID}}fav.user_id = :favUserID{{else}}false{{end}}
 	WHERE true
 	{{if .Omit}}
-		AND not id = any(:omit)
+		AND NOT sched.id = any(:omit)
 	{{end}}
 	{{if .SearchStr}}
 		AND (sched.name ILIKE :search OR sched.description ILIKE :search)
 	{{end}}
 	{{if .After.Name}}
-		AND lower(sched.name) > lower(:afterName)
+		AND
+		{{if not .FavoritesFirst}}
+			lower(sched.name) > lower(:afterName)
+		{{else if .After.IsFavorite}}
+			((fav notnull AND lower(sched.name) > lower(:afterName)) OR fav isnull)
+		{{else}}
+			(fav isnull AND lower(sched.name) > lower(:afterName))
+		{{end}}
 	{{end}}
-	ORDER BY lower(sched.name)
+	ORDER BY {{ .OrderBy }}
 	LIMIT {{.Limit}}
 `))
 
 type renderData SearchOptions
 
+func (opts renderData) OrderBy() string {
+	if opts.FavoritesFirst {
+		return "fav, lower(sched.name)"
```

Comment:
Use `fav isnull` instead of just the row, or sorting won't work correctly for favorites.